R/auto.R

```diff
@@ -23,27 +23,57 @@ pre_release <- function(which = "patch") {
 
 pre_release_impl <- function(which) {
   stopifnot(git2r::is_branch(git2r::repository_head()))
+
+  # FIXME: This fails if the branch is not yet pushed.
+  # Always use the remote of the first remote branch that git2r::branch() finds?
```

Comment:
I think we should push to a remote named "origin", this seems most consistent with usethis workflows.

Reply:
While `"origin"` is usually the default remote name, we are even stricter here by checking the actual remote name. This should also work nicely with multiple remote names, e.g. in forked scenarios?

Should we really hardcode to `"origin"`?

Reply:
Right, let's auto-detect the remote name.

The release branch is always pushed to the same remote as the main branch.